plugins/module_utils/robot.py

```diff
@@ -112,7 +114,11 @@ def raw_plugin_open_url_json(plugin, url, method='GET', timeout=10, data=None, h
     if not content:
         if allow_empty_result and status in allowed_empty_result_status_codes:
             return None, None
-        raise PluginException('Cannot retrieve content from {0}, HTTP status code {1}'.format(url, status))
+        raise PluginException(
+            "Cannot retrieve content from {0} (Error msg: {1}), [{2}] HTTP status code {3}".format(
+                url, reason, method, status
+            )
```

Comment:
`reason` is basically a stringified version of `status`, so it should be printed next to it. Also `method` belongs to `url`:
```suggestion
            "Cannot retrieve content from {0} {1}, HTTP status code {2} ({3})".format(
                method, url, status, reason
            )
```